src/Execution/Arguments/NestedManyToMany.php

```diff
@@ -59,4 +59,33 @@ public function __invoke($parent, $args)
             );
         }
     }
+
+    /**
+     * @param  \Nuwave\Lighthouse\Execution\Arguments\Argument $args
+     *
+     * @return array
+     */
+    private function generateRelationArray(Argument $args)
+    {
+        $values = $args->toPlain();
+
+        if (empty($values)) {
+            return [];
+        }
+
+        if (! is_array($values[0])) {
+            // first values isn't array. Assume values are simply IDs, and return them (old behaviour)
+            return $values;
+        } else {
+            // assume values are arrays and contains pivot information
+            $relationArray = [];
+            foreach ($values as $value) {
+                $id = $value['id'];
+                unset($value['id']);
+                $relationArray[$id] = $value;
```

Comment:
What happens if there are no other values?

According to the example you have shown in your issue passing the `id` as the value would be the way to go.

```php
$person->jobs()->sync([
  1,
  2 => [
    'from' => '...',
    'till' => '...'
  ]
]);
```

We should at least cover this use case of mixing values with just `id` and ones with additional data, and have a test for it.

Reply:
I just read implementation of `sync()` in laravel, if you will pass an array from example above, laravel will convert input to the following array:

```php
[
  1 => [],
  2 => [
    'from' => '...',
    'till' => '...'
  ]
]
```

So in words, it sanitize the input in the way, that final array keys are actually IDs, and the value is an array with values of pivot columns.

I can also add a test that just accepts an id as input without any pivot data